eth/filters/api.go

```diff
@@ -345,6 +347,23 @@ func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([
 		if crit.ToBlock != nil {
 			end = crit.ToBlock.Int64()
 		}
+
+		// if maxBlockRange configured then check for it
+		if api.maxBlockRange != -1 {
+			beginBlock, err := api.backend.HeaderByNumber(ctx, rpc.BlockNumber(begin))
+			if beginBlock == nil || err != nil {
+				return nil, fmt.Errorf("couldn't find fromBlock, fromBlock: %d", begin)
```

Comment:
Overall it looks good. But I'm concerned that this will break some applications in a subtle way, since it's not the same behaviour as before.

Before:

```
$ curl -H 'Content-Type: application/json' --compressed -d '{"jsonrpc":"2.0","id":"id","method":"eth_getLogs", "params": [{ "fromBlock": "0x9999999", "toBlock": "0x100", "topics": [] }]}' http://localhost:8545 | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   188  100    62  100   126   9695  19702 --:--:-- --:--:-- --:--:--  183k
{
  "jsonrpc": "2.0",
  "id": "id",
  "result": []
}
```

Now:

```
$ curl -H 'Content-Type: application/json' --compressed -d '{"jsonrpc":"2.0","id":"id","method":"eth_getLogs", "params": [{ "fromBlock": "0x9999999", "toBlock": "0x100", "topics": [] }]}' http://localhost:8545 | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   246  100   120  100   126  41422  43493 --:--:-- --:--:-- --:--:--  240k
{
  "jsonrpc": "2.0",
  "id": "id",
  "error": {
    "code": -32000,
    "message": "couldn't find fromBlock, fromBlock: 161061273"
  }
}
```

Reply:
sounds reasonable, but I think we still should leave this https://github.com/scroll-tech/go-ethereum/blob/486e5ff35e0a03ac700ff63ce6dc0c926ffaaac4/eth/filters/api.go#L364

Reply:
then I changed logic to following:
if `maxBlockRange` is configured and we can determine `beginblock` and `endblock` without any errors then check for range and return error if not fit, otherwise will work previous logic

Reply:
That still doesn't sound ideal. If I set `fromBlock: 0, toBlock: 99999999` that will bypass this whole restriction and put pressure on the load. I think we should do this check where we already know the actual block numbers, inside `filter`.

Reply:
now check is inside `filter`. Probably we want to limit range for [`eth_getFilterLogs`](https://github.com/scroll-tech/go-ethereum/blob/76d1c1434ffd897c83edf02704233c9ec1b8ffb9/eth/filters/api.go#L382) too